azurerm/resource_arm_media_services.go

```diff
@@ -0,0 +1,194 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/mediaservices/mgmt/2018-07-01/media"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMediaServices() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMediaServicesCreateUpdate,
+		Read:   resourceArmMediaServicesRead,
+		Update: resourceArmMediaServicesCreateUpdate,
+		Delete: resourceArmMediaServicesDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[-a-z0-9]{3,24}$"),
+					"Media Services Account name must be 3 - 24 characters long, contain only lowercase letters and numbers.",
+				),
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"tags": tagsSchema(),
+
+			"storage_account": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"is_primary": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
```

Comment:
Could we be explicit with our default here?

Reply:
Added a default value of false